runtime/src/bank.rs

```diff
@@ -1099,6 +1101,7 @@ pub struct Bank {
 
     /// true when the bank's freezing or destruction has completed
     bank_freeze_or_destruction_incremented: AtomicBool,
+    pub(crate) scheduler: RwLock<InstalledSchedulerBoxInBank>,
```

Comment:
Can you explain the design decision here? It doesn't feel right to me that the scheduler lives on the bank, rather than say `BankWithScheduler`. It seems like most of the complexity in the PR is coming from this decision, but it's unclear to me why it is this way.

Reply:
I tried tracking through where this is used, and didn't see anything that stood out as requiring this to be here.

Maybe https://github.com/solana-labs/solana/pull/31239/files#r1174796706?
Or the complicated drop chain.

Reply:
> Can you explain the design decision here? It doesn't feel right to me that the scheduler lives on the bank, rather than say BankWithScheduler

maybe, this explanation is too terse? or i should have put a pointer comment at `struct Bank {...}`?:

https://github.com/solana-labs/solana/pull/31239#pullrequestreview-1428567713:

```
//! It lends one of pooled `trait InstalledScheduler`s out to a `Bank`, so that the ubiquitous
//! `Arc<Bank>` can conveniently work as a facade for transaction scheduling, to higher-layer
//! subsystems (i.e. both `ReplayStage` and `BankingStage`).
```

Reply:
hey pardon for my too terse reply. it's almost bed time for me tonight... ;) but i wan to get initial impression..:

so, I'm changing my mind. how about this direction?: https://github.com/solana-labs/solana/commit/af0554ebbc41b725d92d61f7782ccab48008e071

if looks good, I'll finishes off the draft commit.

good night. :)

Reply:
hey, I've pushed ci-passing commit series for this big thing: 81075ac, b3db6a034, 3f3dbde, 5c2d4fe

(still, needs some doc update including the nice mermaid chart and a few cleaning, should be reviewable for changed code, though...)